### PR TITLE
Fix date normalization

### PR DESCRIPTION
**Pull request by michamos**

### CODE REVIEW

File: inspire_dojson/hep/rules/bd5xx.py [hunk 1]
```diff
@@ -63,7 +63,13 @@ def _means_not_curated(public_note):
         for public_note in force_list(value.get('a')):
             match = IS_DEFENSE_DATE.match(public_note)
             if match:
-                thesis_info['defense_date'] = normalize_date(match.group('defense_date'))
+                try:
+                    thesis_info['defense_date'] = normalize_date(match.group('defense_date'))
+                except:
```
Comment (jacquerie) [inline inspire_dojson/hep/rules/bd5xx.py]: `except` -> `except Exception`

File: inspire_dojson/hep/rules/bd2xx.py [hunk 2]
```diff
@@ -130,7 +129,7 @@ def imprints2marc(self, key, value):
 
 @hep.over('preprint_date', '^269..')
 def preprint_date(self, key, value):
-    return normalize_date(value.get('c'))
+    return normalize_date_aggressively(value.get('c'))
```
Comment (jacquerie) [inline inspire_dojson/hep/rules/bd2xx.py]: Can you add previously failing tests for these two?

File: inspire_dojson/jobs/rules.py [hunk 3]
```diff
@@ -72,7 +72,7 @@ def _contains_url(val):
             else:
                 self['urls'] = [{'value': closing_info}]
         else:
-            closing_date = normalize_date(closing_info)
+            closing_date = normalize_date_aggressively(closing_info)
```
Comment (jacquerie) [inline inspire_dojson/jobs/rules.py]: Oh, and for these two as well... we have plenty of those : )

**Comment by michamos** [comment]
I added tests for everything and `except`ed the right exception.

**Pull request merged**